f5_openstack_agent/lbaasv2/drivers/bigip/disconnected_service.py

```diff
@@ -31,11 +32,35 @@ def __init__(self, driver):
         self.enabled = (True if self.get_physical_network() else False)
         if self.enabled:
             greenthread.spawn(self.polling_thread)
-        self.start_time = {}
+        self.timer = {}
 
     def get_physical_network(self):
         return self.driver.conf.f5_network_segment_physical_network
 
+    def get_service_virtual_address(self, bigips, virtual):
+        va_table = {}
+        for bigip in bigips:
+            vsf = bigip.tm.ltm.virtuals.virtual
+            vs = vsf.load(
+                name=virtual['name'], partition=virtual['partition'])
+            destination = getattr(vs, 'destination', None)
+            va_table[bigip] = destination
+        return va_table
+
+    def remove_obsolete_service_address(self, bigips, virtual,
+                                        starting_dest, ending_dest):
+        for bigip in bigips:
+            if starting_dest[bigip] != ending_dest[bigip]:
+                vac = bigip.tm.ltm.virtual_address_s
+                dest = basename(starting_dest[bigip])
```

Comment:
Wouldn't this raise an attributeerror if the starting_dest is None, as could be possibly set in line 46 above?